### PR TITLE
Pull 5193

### PR DESCRIPTION
**Pull request by zahiraam**
This is to fix the issue about the BP on windows not being reached. The reason being that the checksum is being calculated for a temp file (the appended file with the footer) that we ultimately don't have access to (unless it's copying in the disk which we don't do). This patch (the code in get CGDebugInfo::getOrCreateFile ) is fixing this issue.

### CODE REVIEW

File: clang/test/CodeGen/debug-info-file-checksum.c [hunk 1]
```diff
@@ -15,4 +15,4 @@
 // Ensure #line directives without name do emit checksums
 // RUN: %clang -emit-llvm -S -g -gcodeview -x c %S/Inputs/debug-info-file-checksum-line.cpp -o - | FileCheck %s --check-prefix CHECKSUM
 
-// CHECKSUM: !DIFile(filename: "{{.*}}debug-info-file-checksum-line.cpp", directory:{{.*}}, checksumkind: CSK_MD5, checksum: "7b568574d0e3c56c28e5e0234d1f4a06")
+// CHECKSUM: !DIFile(filename: "{{.*}}debug-info-file-checksum-line.cpp", directory:{{.*}}, checksumkind: CSK_MD5, checksum: "4456fffb9436b349cea43c698a726b08")
```
Comment (AlexeySachkov) [inline clang/test/CodeGen/debug-info-file-checksum.c:18]: This change doesn't look right, why did checksum change here? I don't see changes to `debug-info-file-checksum-line.cpp` file and therefore I don't expect checksum to be different. Am I missing something?

**Comment by zahiraam** [comment]
@AlexeySachkov Can you please take a look? 
It strange I can re-start the pre-commit and it terminates successfully. But I get an error here. Complaining that I am in a detached HEAD. I checked out your PR, https://github.com/intel/llvm/pull/5193 . May be that's the reason!

**Comment by elizabethandrews** [comment]
Can you please add a description and title for PR?

**Comment by zahiraam** [comment]
> Can you please add a description and title for PR?

@elizabethandrews  Thanks for looking at it. The description is in https://github.com/intel/llvm/pull/5193 
This PR is meant to fix the second part of the bug. I will add it to the description here. Eventually only one PR will be pushed to fix both issues.

File: clang/lib/CodeGen/CGDebugInfo.cpp [hunk 2]
```diff
@@ -406,6 +406,17 @@ llvm::DIFile *CGDebugInfo::getOrCreateFile(SourceLocation Loc) {
 
   SmallString<32> Checksum;
 
+  if (CGM.getCodeGenOpts().SYCLUseMainFileName && FID.isInvalid() &&
+      !llvm::sys::path::is_absolute(FileName)) {
+    FileID MainFileID = SM.getMainFileID();
+    auto ExpectedFileRef = SM.getFileManager().getFileRef(FileName);
```
Comment (stevemerr) [inline clang/lib/CodeGen/CGDebugInfo.cpp:412]: It looks like you could potentially come through this code for some file other than the "main" file.

Comment (zahiraam) [reply clang/lib/CodeGen/CGDebugInfo.cpp:412]: DO you have an example in mind?

Comment (stevemerr) [reply clang/lib/CodeGen/CGDebugInfo.cpp:412]: No, I tried a few but didn't find one that would trigger this.

Comment (zahiraam) [reply clang/lib/CodeGen/CGDebugInfo.cpp:412]: I might potentially put this under a condition  like this: if(SM.getFileEntryForID(SM.getMainFileID())) I suppose, or add an assert to make sure that we catch it if it happens.

**Pull request closed without merging**